src/jailer/src/env.rs

```diff
@@ -247,6 +238,29 @@ impl Env {
         self.uid
     }
 
+    fn validate_exec_file(exec_file: &str) -> Result<(PathBuf, String)> {
+        let exec_file_path = canonicalize(exec_file)
+            .map_err(|err| Error::Canonicalize(PathBuf::from(exec_file), err))?;
+
+        if !exec_file_path.is_file() {
+            return Err(Error::NotAFile(exec_file_path));
+        }
+
+        let exec_file_name = exec_file_path
+            .file_name()
+            .ok_or_else(|| Error::FileName(exec_file_path.clone()))?
```

Comment:
I think we should rename `FileName` error to `ExtractFileName` or something similar. It's confusing to keep it as is because we introduced a second error for the file name: `ExecFileName`. `FileName` it's too general and it's hard to tell what the error is about.